### PR TITLE
Fix method return

### PR DESCRIPTION
**Pull request by lattice0**
fixes https://github.com/fzyzcjy/flutter_rust_bridge/issues/585, that is, when we return a struct with a method in a non method function, it should work

## Checklist

- [x] New tests are added to ensure new features are working (probably by modifying the `frb_example/pure_dart` example).
- [x] All existing and new tests are passing.
- [x] If this PR adds/changes features, documentations (in `book` folder) are modified as well.

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Good job! I guess need to run code formatter to make CI happy

File: frb_codegen/src/generator/dart/mod.rs [hunk 1]
```diff
@@ -403,7 +408,11 @@ fn generate_api_func(func: &IrFunc, ir_file: &IrFile) -> GeneratedApiFunc {
                 "".to_string()
             }
         })
+        // If struct has a method with first element `input0`
         || (input_0.is_some() && MethodNamingUtil::struct_has_methods(ir_file, input_0.unwrap()))
+        //If output is a struct with methods
+        || (func_output_struct_name.is_some()
+            && MethodNamingUtil::has_methods(&func_output_struct_name.unwrap(), ir_file))
```
Comment (fzyzcjy) [inline frb_codegen/src/generator/dart/mod.rs]: nit: Shall we merge `has_methods` and `struct_has_methods` functions? They look a bit similar. (method_utils.rs)

Comment (lattice0) [reply frb_codegen/src/generator/dart/mod.rs]: They use different techniques. One does a look on the name and also searches using the method name, the other searches for the first argument receiving a self to the struct type

Comment (fzyzcjy) [reply frb_codegen/src/generator/dart/mod.rs]: I see. That is partially why I think it may be a bit confusing...

Comment (lattice0) [reply frb_codegen/src/generator/dart/mod.rs]: removed!

**Comment by fzyzcjy** [comment]
Now wait for CI and merge when it goes green :)

**Comment by lattice0** [comment]
Can you rerun? this error is strange

**Comment by fzyzcjy** [comment]
sure

**Comment by fzyzcjy** [comment]
No worries about that, that did occur before (though very weird)

**Pull request merged**